### DISCUSSION TITLE
Documentation Clarification - Where can variables be substituted

### DISCUSSION THREAD

**Discussion by mattharrigan** [argoproj/argo-workflows#9647] [General]
Hello,

I have read [https://argoproj.github.io/argo-workflows/variables/](https://argoproj.github.io/argo-workflows/variables/). My question is what fields in the spec can use variables? For example in a WorkflowTemplate, can the image, or a secret name, or configmap name use `"{{inputs.parameters.something}}"`? This might be a good addition to the docs.

Thanks for a really useful project,
Matt

**Comment by rdean150** [comment]
Another specific example: Can the "name" and "depends" fields of DAG tasks accept and interpolate variables/tags? 

The answer to that question is hugely important for teams who may prefer to generate their workflows dynamically or maintain their task relationships in a database instead of a yaml file. Or even simply import it from an external json file.

And to echo OP, thanks for all the hard work with this project. Great stuff!
